packages/@webex/plugin-meetings/src/multistream/remoteMediaManager.ts

```diff
@@ -525,33 +543,96 @@ export class RemoteMediaManager extends EventsScope {
     this.mediaRequestManagers.video.commit();
   }
 
+  /**
+   * Sets which named media group need receiving
+   * @param {MediaType} mediaType of the stream
+   * @param {number} languageCode of the stream
+   * @returns {void}
+   */
+  public async setReceiveNamedMediaGroup(mediaType: MediaType, languageId: number) {
+    if (mediaType !== MediaType.AudioMain) {
+      throw new Error(`cannot set receive named media group which media type is ${mediaType}`);
+    }
+
+    const value = languageId;
+    if (value === this.config.namedMediaGroup?.value) {
+      return;
+    }
+
+    this.config.namedMediaGroup = {
+      type: 1,
+      value,
+    };
+
+    await this.createInterpretationAudioMedia(false);
+
+    this.media.audio.main?.sendActiveSpeakerMediaRequest(true);
+  }
+
   /**
    * Creates the audio slots
    */
   private async createAudioMedia() {
-    // create the audio receive slots
+    // create si audio request
+    await this.createInterpretationAudioMedia(false);
+
+    // create main audio receive slots
     for (let i = 0; i < this.config.audio.numOfActiveSpeakerStreams; i += 1) {
       // eslint-disable-next-line no-await-in-loop
       const slot = await this.receiveSlotManager.allocateSlot(MediaType.AudioMain);
 
-      this.slots.audio.push(slot);
+      this.slots.audio.main.push(slot);
     }
 
-    // create a remote media group
-    this.media.audio = new RemoteMediaGroup(
+    // create a remote media group for main audio
+    this.media.audio.main = new RemoteMediaGroup(
       this.mediaRequestManagers.audio,
-      this.slots.audio,
+      this.slots.audio.main,
       255,
       true
     );
 
     this.emit(
       {file: 'multistream/remoteMediaManager', function: 'createAudioMedia'},
       Event.AudioCreated,
-      this.media.audio
+      this.media.audio.main
     );
   }
 
+  /**
+   * Creates the audio slots for named media
+   */
+  private async createInterpretationAudioMedia(commitRequest: boolean) {
+    // release existing si slot
+    if (this.slots.audio.si) {
+      this.receiveSlotManager.releaseSlot(this.slots.audio.si);
```

Comment:
createAudioMedia() only allocates the slots, doesn't release them, but createInterpretationAudioMedia() also releases the existing slot, so this is inconsistent. We should either have a separate create/destroy functions for si or rename this one to "recreate" or something like that? I think I prefer the 1st option.

Actually, after you do the changes from my other comments, you won't need to release the slot at all, right? because si audio will be created once at the start and the same one will be re-used all the time. Unless you want to destroy it completely when user no longer wants to receive any si audio, then we would probably also need to emit a new "media:remoteInterpretationAudio:destroyed" event

Reply:
Rmoved destroy part